ui/views.py

```diff
@@ -84,69 +86,168 @@ class VideoDetail(TemplateView):
     """
     template_name = "ui/video_detail.html"
 
-    def get_context_data(self, video_key, **kwargs):  # pylint: disable=arguments-differ
+    def get(self, request, *args, **kwargs):
```

Comment:
This function may be useful: https://docs.djangoproject.com/en/2.0/topics/auth/default/#django.contrib.auth.decorators.user_passes_test

It takes a function which, depending on the return value, will say whether or not the user can see the page. If not the user is redirected to the login

Reply:
Thanks @noisecapella !

Reply:
That decorator just takes user as argument, I'd need to pass in the video as well.  So I placed the conditional redirect logic in a function instead.